spec/models/manageiq/providers/nuage/network_manager/vcr_specs/refresher_spec.rb

```diff
@@ -44,5 +70,129 @@ def assert_table_counts
+      .to match_array([cloud_subnet_ref1, cloud_subnet_ref2])
+  end
+
+  def assert_network_groups
+    g1 = NetworkGroup.find_by(:ems_ref => network_group_ref1)
+    expect(g1).to have_attributes(
+      :name                   => "Ansible-Test",
+      :cidr                   => nil,
+      :status                 => "active",
+      :enabled                => nil,
+      :ems_id                 => @ems.id,
+      :orchestration_stack_id => nil,
+      :type                   => "ManageIQ::Providers::Nuage::NetworkManager::NetworkGroup"
+    )
+    expect(g1.cloud_subnets.count).to eq(0)
+    expect(g1.security_groups.count).to eq(0)
+
+    g2 = NetworkGroup.find_by(:ems_ref => network_group_ref2)
+    expect(g2).to have_attributes(
+      :name                   => "XLAB",
+      :cidr                   => nil,
+      :status                 => "active",
+      :enabled                => nil,
+      :ems_id                 => @ems.id,
+      :orchestration_stack_id => nil,
+      :type                   => "ManageIQ::Providers::Nuage::NetworkManager::NetworkGroup"
+    )
+    expect(g2.cloud_subnets.count).to eq(2)
+    expect(g2.security_groups.count).to eq(1)
+
+    expect(g2.cloud_subnets.map(&:ems_ref))
+      .to match_array([cloud_subnet_ref1, cloud_subnet_ref2])
+    expect(g2.security_groups.map(&:ems_ref))
+      .to match_array([security_group_ref])
+  end
+
+  def assert_security_groups
+    g1 = SecurityGroup.find_by(:ems_ref => security_group_ref)
+    expect(g1).to have_attributes(
+      :name                   => "Test Policy Group",
+      :description            => nil,
+      :type                   => "ManageIQ::Providers::Nuage::NetworkManager::SecurityGroup",
+      :ems_id                 => @ems.id,
+      :cloud_network_id       => nil,
+      :cloud_tenant_id        => nil,
+      :orchestration_stack_id => nil
+    )
+    expect(g1.network_group.ems_ref).to eq(network_group_ref2)
+  end
+
+  def assert_cloud_subnets
+    s1 = CloudSubnet.find_by(:ems_ref => cloud_subnet_ref1)
+    expect(s1).to have_attributes(
+      :name                           => "Subnet 1",
+      :ems_id                         => @ems.id,
+      :availability_zone_id           => nil,
+      :cloud_network_id               => nil,
+      :cidr                           => "10.10.20.0/24",
+      :status                         => nil,
+      :dhcp_enabled                   => false,
+      :gateway                        => "10.10.20.1",
+      :network_protocol               => "ipv4",
+      :cloud_tenant_id                => nil,
+      :dns_nameservers                => nil,
+      :ipv6_router_advertisement_mode => nil,
+      :ipv6_address_mode              => nil,
+      :type                           => "ManageIQ::Providers::Nuage::NetworkManager::CloudSubnet",
+      :network_router_id              => nil,
+      :network_group_id               => NetworkGroup.find_by(:ems_ref => network_group_ref2).id,
```

Comment:
Can you use `@data_index.fetch_path` here as well?

Reply:
I don't think I have access to refresh parser internals here in the spec. Also I think the spec should really check database status after the refresh, not the parser internals.